### ISSUE TITLE
Certificate with same Distinguished Name have different behaviour (PKI)

### ISSUE THREAD

**Issue by CiraciNicolo** [hashicorp/vault#5865]
**Describe the bug**
I'm generating a CSR via Windows Security API to be signed by the PKI service but I get the following error if the singing process is requested via sign-verbatim REST endpoint:

> "1 error occurred:\n\t* unable to create certificate: asn1: cannot marshal nil value\n\n"

This error doesn't appear if the certificate **with the same Distinguished Name** is generated via openssl

**To Reproduce**
1. Generate a CSR with Windows Security .NET API
2. Use REST to sign-verbatin the CSR
3. See error

**Expected behavior**
I expect consistency between the two csr.

**Environment:**
* Version 0.11.5
* Vault v0.11.5 ('a59ffa4a0f09bbf198241fe6793a96722789b639')
* Docker on Fedora 29

Vault server configuration file(s):

```json
{
	"listener": [{
		"tcp": {
			"address": "0.0.0.0:8200",
			"tls_disable" : 1
		}
	}],
	"storage" :{
		"file" : {
			"path" : "/vault/data"
		}
	},
	"ui": true,
	"max_lease_ttl": "10h",
	"default_lease_ttl": "10h"
}

```

**Comment by CiraciNicolo** [comment]
So, I was able to pinpoint the issue. It appears that Microsoft APIs screw up something if the CN is an email address (I need the certificates for identity management). CSRs generated by openssl with an email address as CN are working fine.

We can close this issue if you are not interested in this kind of issue.

**Comment by jefferai** [comment]
It's less "not interested" and more "can't do much about". I'd still be happy to take a look and see if it's something that can be worked around in Vault or can be filed upstream in Go for a workaround, but I'd need those CSRs.

**Comment by CiraciNicolo** [comment]
Actually I was able to fix the issue in a matter of minutes once I got to understand the problem, it seems that I needed to force UTF8 encoding using the flag `X500DistinguishedNameFlags.ForceUTF8Encoding`. Sorry for the trouble, maybe this can help someone else. 

What maybe would be great is more 'talking' errors, maybe if TRACE log mode is enable but I also understand that this is a shared library (or even core go) so it is not trivial.
